File: lib/src/app/login/page.js

"use client";
import { useState, useEffect } from "react";
import { Mail, Lock, ArrowRight, Loader2 } from "lucide-react";
import { useRouter } from "next/navigation";

import apiConfig from "@/utils/apiConfig";

const ONBOARDING_DATA = [
  {
    image: "https://images.unsplash.com/photo-1595079676339-1534801ad6cf?auto=format&fit=crop&q=80&w=1200",
    title: "Simple QR Access",
    description: "Customers scan a unique QR code at their table to instantly access your digital storefront."
  },
  {
    image: "https://images.unsplash.com/photo-1556742044-3c52d6e88c62?auto=format&fit=crop&q=80&w=1200",
    title: "Browse the Menu",
    description: "A beautiful, interactive menu on their own device. No more waiting for physical menus."
  },
  {
    image: "https://images.unsplash.com/photo-1516321318423-f06f85e504b3?auto=format&fit=crop&q=80&w=1200",
    title: "Instant Ordering",
    description: "Orders go directly from the customer's phone to your kitchen. Speed up service and reduce errors."
  },
  {
    image: "https://images.unsplash.com/photo-1460925895917-afdab827c52f?auto=format&fit=crop&q=80&w=1200",
    title: "Manage with Ease",
    description: "Watch orders arrive in real-time on your dashboard. Track sales and manage inventory effortlessly."
  }
];

export default function LoginPage() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [loading, setLoading] = useState(false);
  const [activeSlide, setActiveSlide] = useState(0);

  useEffect(() => {
    const timer = setInterval(() => {
      setActiveSlide((prev) => (prev + 1) % ONBOARDING_DATA.length);
    }, 5000);
    return () => clearInterval(timer);
  }, []);

  // Designer Tip: Button lights up only if email has @ and password is typed
  const isFormValid = email.includes("@") && password.length > 0;

  const router = useRouter();

  const handleLogin = async (e) => {
    e.preventDefault();
    if (loading) return;

    setLoading(true);

    try {
      const res = await fetch(`${apiConfig?.BASE_URL}/api/auth/login`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        credentials: "include",
        body: JSON.stringify({ email, password }),
      });

      const data = await res.json();

      if (!res.ok)
        throw new Error(data.message || "Login failed");

      // 🔥 SAVE SESSION (for both middleware and client-side libs)
      if (data.token) {
        document.cookie = `token=${data.token}; path=/; max-age=${24 * 60 * 60}; SameSite=Lax`;
        
        // Use existing utility if available
        import("@/app/lib/auth").then(({ saveAuthSession }) => {
          saveAuthSession(data);
        }).catch(() => {
          // Fallback if import fails
          localStorage.setItem("qr_serve_session", JSON.stringify({
            token: data.token,
            role: data.role,
            restaurantId: data.restaurantId || data.restaurants?.[0]?._id,
            userName: data.name || "Admin"
          }));
        });
      }

      window.location.href = "/dashboard/orders";
    } catch (err) {
      alert(err.message);
    } finally {
      setLoading(false);
    }
  };


  return (
    <div className="min-h-screen bg-white flex flex-col lg:flex-row font-['Inter',sans-serif]">
      <section className="relative hidden lg:flex lg:w-1/2 h-screen sticky top-0 overflow-hidden">
        {ONBOARDING_DATA.map((slide, index) => (
          <div key={index} className={`absolute inset-0 transition-opacity duration-1000 ${activeSlide === index ? "opacity-100" : "opacity-0"}`}>
            <img src={slide.image} className="w-full h-full object-cover" alt="" />
            <div className="absolute inset-0 bg-black/40" />
            <div className="absolute bottom-0 left-0 p-16 w-full z-10">
              <h2 className="text-5xl font-extrabold text-white-force mb-4">{slide.title}</h2>
              <p className="text-xl text-white-force font-medium opacity-90">{slide.description}</p>
            </div>
          </div>
        ))}
      </section>

      <section className="flex-1 flex items-center justify-center p-6 sm:p-12 bg-white overflow-y-auto">
        <div className="max-w-[420px] w-full py-10">
          <div className="mb-8 flex items-center gap-2">
            <div className="w-10 h-10 bg-[#FF5C00] rounded-xl flex items-center justify-center">
              <span className="text-white-force font-black text-xl">Q</span>
            </div>
            <span className="text-2xl font-black text-slate-900 tracking-tighter">QRserve</span>
          </div>

          <h2 className="text-3xl font-extrabold text-slate-900 mb-2">Welcome Back</h2>
          <p className="text-slate-500 font-medium mb-10">Access your admin control center.</p>

          <form className="space-y-6" onSubmit={handleLogin}>
            <div className="flex flex-col gap-[6px]">
              <label className="text-[11px] font-extrabold text-slate-400 uppercase tracking-widest ml-1">Email Address</label>
              <div className="relative group">
                <Mail className="absolute left-4 top-1/2 -translate-y-1/2 text-slate-400" size={18} />
                <input
                  placeholder="admin@qrserve.com"
                  className="w-full bg-slate-50 border border-slate-200 rounded-2xl py-4 pl-12 focus:border-[#FF5C00] outline-none font-semibold text-slate-800 placeholder:text-slate-400"
                  onChange={e => setEmail(e.target.value)}
                />
              </div>
            </div>

            <div className="flex flex-col gap-[6px]">
              <label className="text-[11px] font-extrabold text-slate-400 uppercase tracking-widest ml-1">Secret Password</label>
              <div className="relative group">
                <Lock className="absolute left-4 top-1/2 -translate-y-1/2 text-slate-400" size={18} />
                <input
                  type="password" placeholder="••••••••"
                  className="w-full bg-slate-50 border border-slate-200 rounded-2xl py-4 pl-12 focus:border-[#FF5C00] outline-none font-semibold text-slate-800 placeholder:text-slate-400"
                  onChange={e => setPassword(e.target.value)}
                />
              </div>
            </div>

            <button
              disabled={!isFormValid || loading}
              className={`w-full py-5 rounded-2xl transition-all flex items-center justify-center gap-3 mt-4 font-bold uppercase tracking-tight
                ${!isFormValid ? "bg-slate-100 text-slate-400 cursor-not-allowed" : "bg-[#FF5C00] text-white-force hover:bg-[#E65200] active:scale-[0.98]"}`}
            >
              {loading ? <Loader2 className="animate-spin" /> : <>Enter Dashboard <ArrowRight size={20} /></>}
            </button>
          </form>
          {/* ADD THIS BLOCK BELOW */}
          <div className="mt-10 pt-8 border-t border-slate-100 text-center">
            <p className="text-slate-400 font-medium text-sm">
              Don't have an account?{" "}
              <a href="/signup" className="text-[#FF5C00] font-bold hover:underline underline-offset-4">
                Sign Up
              </a>
            </p>
          </div>
        </div>
      </section>
    </div>
  );
}

File: src/app/login/page.dart
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:lucide_icons/lucide_icons.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import '../../utils/apiClient.dart';
import '../context/AuthContext.dart';

class LoginPage extends ConsumerStatefulWidget {
  const LoginPage({super.key});

  @override
  ConsumerState<LoginPage> createState() => _LoginPageState();
}

class _LoginPageState extends ConsumerState<LoginPage> {
  bool _isLoading = false;
  int _activeSlide = 0;
  Timer? _timer;

  String _email = '';
  String _password = '';

  final List<Map<String, String>> _onboardingData = [
    {
      "image":
          "https://images.unsplash.com/photo-1595079676339-1534801ad6cf?auto=format&fit=crop&q=80&w=1200",
      "title": "Simple QR Access",
      "description":
          "Customers scan a unique QR code at their table to instantly access your digital storefront.",
    },
    {
      "image":
          "https://images.unsplash.com/photo-1556742044-3c52d6e88c62?auto=format&fit=crop&q=80&w=1200",
      "title": "Browse the Menu",
      "description":
          "A beautiful, interactive menu on their own device. No more waiting for physical menus.",
    },
    {
      "image":
          "https://images.unsplash.com/photo-1516321318423-f06f85e504b3?auto=format&fit=crop&q=80&w=1200",
      "title": "Instant Ordering",
      "description":
          "Orders go directly from the customer's phone to your kitchen. Speed up service and reduce errors.",
    },
    {
      "image":
          "https://images.unsplash.com/photo-1460925895917-afdab827c52f?auto=format&fit=crop&q=80&w=1200",
      "title": "Manage with Ease",
      "description":
          "Watch orders arrive in real-time on your dashboard. Track sales and manage inventory effortlessly.",
    },
  ];

  @override
  void initState() {
    super.initState();
    _timer = Timer.periodic(const Duration(seconds: 5), (timer) {
      if (mounted) {
        setState(() {
          _activeSlide = (_activeSlide + 1) % _onboardingData.length;
        });
      }
    });
  }

  @override
  void dispose() {
    _timer?.cancel();
    super.dispose();
  }

  bool get _isFormValid => _email.contains('@') && _password.isNotEmpty;

  Future<void> _handleLogin() async {
    if (!_isFormValid || _isLoading) return;

    setState(() => _isLoading = true);

    try {
      final res = await apiFetch(
        '/api/auth/login',
        method: 'POST',
        data: {'email': _email, 'password': _password},
      );
      print("🔍 Login Response: $res");

      // Directly update the context with the login response
      // Robust check: if we got here without exception, and we have data/token, it's a success
      final userData = res['data'] ?? res;
      final bool looksSubstantial = (userData is Map && (userData.containsKey('token') || userData.containsKey('restaurantId')));
      print("📝 Evaluation: Success=${res['success']}, LooksSubstantial=$looksSubstantial");

      if (res['success'] == true || (res is Map && looksSubstantial)) {
        print("✅ Login Successful, updating user data");
        ref.read(authProvider.notifier).setUserData(userData is Map<String, dynamic> ? userData : res);
      } else {
        print("⚠️ Login response ambiguous, reloading session");
        // Fallback: reload session if response didn't contain full user info
        await ref.read(authProvider.notifier).loadSession();
      }

      if (mounted) {
        print("🚀 Redirecting to dashboard");
        context.go('/dashboard/orders');
      }
    } catch (e) {
      print("❌ Login Failed Error: $e");
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Login failed: $e'), backgroundColor: Colors.red),
        );
      }
    } finally {
      if (mounted) setState(() => _isLoading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    final authState = ref.watch(authProvider);
    if (authState.user != null) {
      WidgetsBinding.instance.addPostFrameCallback((_) {
        if (mounted) context.go('/dashboard/orders');
      });
    }

    return Scaffold(
      backgroundColor: Colors.white,
      body: Row(
        children: [
          // Carousel Section (Hidden on small screens)
          if (1.sw >= 900)
            Expanded(
              child: Stack(
                fit: StackFit.expand,
                children: [
                  AnimatedSwitcher(
                    duration: const Duration(seconds: 1),
                    child: Image.network(
                      _onboardingData[_activeSlide]['image']!,
                      key: ValueKey<int>(_activeSlide),
                      fit: BoxFit.cover,
                      width: double.infinity,
                      height: double.infinity,
                    ),
                  ),
                  Container(color: Colors.black.withAlpha(100)), // Overlay
                  Positioned(
                    bottom: 0,
                    left: 0,
                    right: 0,
                    child: Padding(
                      padding: EdgeInsets.all(64.0.r),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text(
                            _onboardingData[_activeSlide]['title']!,
                            style: TextStyle(
                              fontSize: 48.sp,
                              fontWeight: FontWeight.w900,
                              color: Colors.white,
                            ),
                          ),
                          SizedBox(height: 16.h),
                          Text(
                            _onboardingData[_activeSlide]['description']!,
                            style: TextStyle(
                              fontSize: 20.sp,
                              color: Colors.white70,
                              fontWeight: FontWeight.w500,
                            ),
                          ),
                          SizedBox(height: 32.h),
                          Row(
                            children: List.generate(_onboardingData.length, (
                              index,
                            ) {
                              return AnimatedContainer(
                                duration: const Duration(milliseconds: 300),
                                margin: EdgeInsets.only(right: 8.w),
                                height: 6.h,
                                width: _activeSlide == index ? 32.w : 8.w,
                                decoration: BoxDecoration(
                                  color: _activeSlide == index
                                      ? const Color(0xFFFF5C00)
                                      : Colors.white38,
                                  borderRadius: BorderRadius.circular(4.r),
                                ),
                              );
                            }),
                          ),
                        ],
                      ),
                    ),
                  ),
                ],
              ),
            ),

          // Form Section
          Expanded(
            child: SingleChildScrollView(
              child: Padding(
                  padding: EdgeInsets.symmetric(
                    horizontal: 24.0.w,
                    vertical: 1.sh < 700 ? 24.0.h : 48.0.h,
                  ),
                child: Center(
                  child: Container(
                    constraints: BoxConstraints(maxWidth: 420.w),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: [
                            Container(
                              width: 40.w,
                              height: 40.w,
                              decoration: BoxDecoration(
                                color: const Color(0xFFFF5C00),
                                borderRadius: BorderRadius.circular(12.r),
                              ),
                              alignment: Alignment.center,
                              child: Text(
                                'Q',
                                style: TextStyle(
                                  color: Colors.white,
                                  fontWeight: FontWeight.w900,
                                  fontSize: 20.sp,
                                ),
                              ),
                            ),
                            SizedBox(width: 8.w),
                            Text(
                              'QRserve',
                              style: TextStyle(
                                fontSize: 24.sp,
                                fontWeight: FontWeight.w900,
                                color: const Color(0xFF0F172A),
                                letterSpacing: -1,
                              ),
                            ),
                          ],
                        ),
                        SizedBox(height: 32.h),
                        Text(
                          'Welcome Back',
                          style: TextStyle(
                            fontSize: 32.sp,
                            fontWeight: FontWeight.w900,
                            color: const Color(0xFF0F172A),
                          ),
                        ),
                        SizedBox(height: 8.h),
                        const Text(
                          'Access your admin control center.',
                          style: TextStyle(
                            color: Colors.grey,
                            fontWeight: FontWeight.w600,
                          ),
                        ),
                        SizedBox(height: 40.h),

                        // Form Fields
                        _buildLabel('Email Address'),
                        _buildTextField(
                          LucideIcons.mail,
                          'admin@qrserve.com',
                          (val) => _email = val,
                          false,
                        ),

                        _buildLabel('Secret Password'),
                        _buildTextField(
                          LucideIcons.lock,
                          '••••••••',
                          (val) => _password = val,
                          true,
                        ),

                        SizedBox(height: 16.h),
                        SizedBox(
                          width: double.infinity,
                          height: 56.h,
                          child: ElevatedButton(
                            onPressed: _isFormValid && !_isLoading
                                ? _handleLogin
                                : null,
                            style: ElevatedButton.styleFrom(
                              backgroundColor: const Color(0xFFFF5C00),
                              disabledBackgroundColor: Colors.grey.shade200,
                              shape: RoundedRectangleBorder(
                                borderRadius: BorderRadius.circular(16.r),
                              ),
                            ),
                            child: _isLoading
                                ? const CircularProgressIndicator(
                                    color: Colors.white,
                                  )
                                : Row(
                                    mainAxisAlignment: MainAxisAlignment.center,
                                    children: [
                                      Text(
                                        'Enter Dashboard',
                                        style: TextStyle(
                                          fontWeight: FontWeight.w900,
                                          fontSize: 14.sp,
                                          color: _isFormValid
                                              ? Colors.white
                                              : Colors.grey.shade400,
                                          letterSpacing: 1,
                                        ),
                                      ),
                                      SizedBox(width: 8.w),
                                      Icon(
                                        LucideIcons.arrowRight,
                                        size: 20.sp,
                                        color: _isFormValid
                                            ? Colors.white
                                            : Colors.grey.shade400,
                                      ),
                                    ],
                                  ),
                          ),
                        ),

                        SizedBox(height: 40.h),
                        Center(
                          child: Wrap(
                            crossAxisAlignment: WrapCrossAlignment.center,
                            children: [
                              const Text(
                                'Don\'t have an account? ',
                                style: TextStyle(
                                  color: Colors.grey,
                                  fontWeight: FontWeight.w600,
                                ),
                              ),
                              InkWell(
                                onTap: () => context.go('/signup'),
                                child: Text(
                                  'Sign Up',
                                  style: TextStyle(
                                    color: const Color(0xFFFF5C00),
                                    fontWeight: FontWeight.w900,
                                    fontSize: 14.sp,
                                  ),
                                ),
                              ),
                            ],
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildLabel(String text) {
    return Padding(
      padding: EdgeInsets.only(left: 4.w, bottom: 8.h),
      child: Text(
        text.toUpperCase(),
        style: TextStyle(
          fontSize: 11.sp,
          fontWeight: FontWeight.w900,
          color: Colors.grey,
          letterSpacing: 1.5,
        ),
      ),
    );
  }

  Widget _buildTextField(
    IconData icon,
    String hint,
    Function(String) onChanged,
    bool isPassword,
  ) {
    return Container(
      margin: EdgeInsets.only(bottom: 24.h),
      decoration: BoxDecoration(
        color: const Color(0xFFF8FAFC),
        borderRadius: BorderRadius.circular(16.r),
        border: Border.all(color: Colors.grey.shade200),
      ),
      child: TextField(
        obscureText: isPassword,
        onChanged: onChanged,
        style: TextStyle(
          fontWeight: FontWeight.w600,
          color: const Color(0xFF334155),
          fontSize: 14.sp,
        ),
        decoration: InputDecoration(
          hintText: hint,
          hintStyle: const TextStyle(color: Colors.grey),
          prefixIcon: Icon(icon, color: Colors.grey, size: 20.sp),
          border: InputBorder.none,
          contentPadding: EdgeInsets.symmetric(vertical: 18.h),
        ),
      ),
    );
  }
}
